middleware/tls.go

```diff
@@ -0,0 +1,26 @@
+package middleware
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+)
+
+
+func NewTLSConfig(certPath, keyPath, caPath string) (*tls.Config, error) {
```

Comment:
Sure. This also should be tweaked to allow clients to specify nothing and use the system CAs with no client cert.